### PR TITLE
Denham/frontend

### PR DESCRIPTION
**Pull request by DenhamPreen**
The join pool button will change to withdraw once its loaded that you have dai in the pool

### CODE REVIEW

File: src/utils/useWeb3Connect.js [hunk 1]
```diff
@@ -170,9 +174,15 @@ function useWeb3Connect() {
     const daiContract = new web3Inited.eth.Contract(daiAbi.abi, DAI_ADDRESS);
     setDaiContract(daiContract);
 
+    // console.log('daiContract');
```
Comment (JasoonS) [inline src/utils/useWeb3Connect.js]: I think we can delete this now that we use the updateBalance function below.

File: src/utils/useWeb3Connect.js [hunk 2]
```diff
@@ -326,23 +343,29 @@ function useWeb3Connect() {
   };
   const updateBalance = async (
```
Comment (JasoonS) [inline src/utils/useWeb3Connect.js]: More a personal preference I guess, but this function is used before it is defined.

File: src/utils/useWeb3Connect.js [hunk 3]
```diff
@@ -387,10 +424,10 @@ function useWeb3Connect() {
       );
       // console.log({ iteration, address });
       let tempCurrentVote = 0;
-      if (connected && address) {
+      if (connected && addr) {
```
Comment (JasoonS) [inline src/utils/useWeb3Connect.js]: Any reason to change this?

Comment (joaoaguiam) [reply src/utils/useWeb3Connect.js]: The reason is that the fetch proposals is now also called from the initialization function. It was relying on the state variable daoContractReadOnly that is set on that function. The setDaoContractReadOnly is not updating the state on the same loop, so the value on this function was undefined. Therefore we send the contract from that init function until this call. 

I know that this is not beautiful, but I struggled with this file and flow...

File: src/utils/useWeb3Connect.js (hunk 1 above)
Comment (joaoaguiam) [reply src/utils/useWeb3Connect.js]: sure :)

**Pull request merged**